server.js

```diff
@@ -1,20 +1,143 @@
+
+const {
+  getEpisodeInfo,
+  getCurrentUserProfile,
+} = require("./lib/SpotifyMethods");
+const {
+  getAppAccessToken,
+  getUserAccessToken,
+  refreshUserAccessToken,
+} = require("./lib/SpotifyAccess");
 
 const app = express();
 const port = process.env.PORT || 3001;
 
-app.get("/oauth/apptoken", async (request, response) => {
-  const apptoken = await getAppAccessToken();
-  response.send(apptoken);
+app.use(cookieParser());
+
+// OAuth Spotify
+
+// Authorization Code Flow
+
+app.get("/oauth/spotify/authorize", (request, response) => {
+  const url =
+    "https://accounts.spotify.com/authorize?" +
+    querystring.stringify({
+      response_type: "code",
+      client_id: process.env.CLIENT_ID,
+      scope: "user-read-private user-read-email",
+      redirect_uri: process.env.REDIRECT_URI,
+    });
+  response.redirect(url);
+});
+
+app.get("/oauth/spotify/validate", async (request, response) => {
+  const { code } = request.query;
+  const {
+    access_token: accessToken,
+    expires_in: expiresIn,
+    refresh_token: refreshToken,
+  } = await getUserAccessToken(code);
+  response.cookie("access", accessToken, {
+    maxAge: (expiresIn - 60) * 1000,
+  });
+  response.cookie("refresh", refreshToken);
+  response.redirect("/vote");
+});
+
+app.get("/oauth/spotify/logout", async (request, response) => {
+  response.clearCookie("access");
+  response.clearCookie("refresh");
+  response.status(200).send("Cookies cleared");
+});
+
+app.get("/oauth/spotify/refreshtoken", async (request, response) => {
+  const refreshToken = request.cookies.refresh;
+  try {
+    const {
+      access_token: newAccessToken,
+      expires_in: expiresIn,
+      refresh_token: newRefreshToken,
+    } = await refreshUserAccessToken(refreshToken);
+    response.cookie("access", newAccessToken, {
+      maxAge: (expiresIn - 60) * 1000,
+    });
+    if (newRefreshToken) {
+      response.cookie("refresh", newRefreshToken);
+    }
+  } catch (error) {
+    response.redirect("/oauth/spotify/authorize");
+    console.error(error);
+  }
+});
+
+// Client Credentials Flow
+
+app.get("/oauth/spotify/apptoken", async (request, response) => {
+  const {
+    access_token: appToken,
+    expires_in: expiresIn,
+  } = await getAppAccessToken();
+  response.cookie("apptoken", appToken, {
+    maxAge: (expiresIn - 60) * 1000,
+  });
+  response.status(200).send("App Token Cookie is set!");
+});
+
+// Spotify API
+
+app.get("/api/user/profile", async (request, response) => {
+  const accessToken = request.cookies.access;
+  const refreshToken = request.cookies.refresh;
+  if (!accessToken && !refreshToken) {
+    response.redirect("/oauth/spotify/authorize");
+    return;
+  }
+  if (!accessToken) {
+    const {
+      access_token: newAccessToken,
+      expires_in: expiresIn,
+      refresh_token: newRefreshToken,
+    } = await refreshUserAccessToken(refreshToken);
+
+    response.cookie("access", newAccessToken, {
+      maxAge: (expiresIn - 60) * 1000,
+    });
```

Comment:
did you use, 3 different cookies? in /api/episode/:id you use/ set a cookie named "apptoken", here you set a cookie named access... in line 94/95 is the third one

````js
const accessToken = request.cookies.access;
  const refreshToken = request.cookies.refresh;
````